### PR TITLE
Feature/run client tests with server

### LINKED ISSUE #728: Integrate client_2sgs in Jenkins automation
Right now we are only executing the client_sg suite, we need to bundle the client_2sgs suite somehow

2 options
- move out p2p tests and execute from the root of client shared tests (probably better)
- put client_2sg tests in client_sg (reprovision topology)

### LINKED ISSUE #760: Reenable longpoll timeout tests for mac
reenable when this is fixed - https://github.com/couchbase/couchbase-lite-ios/issues/1236

two tests
- test_longpoll_changes_termination_timeout
- test_longpoll_changes_termination_heartbeat

### LINKED ISSUE #802: Run client tests against CBS with di mode

### PR DESCRIPTION
**Pull request by sethrosetter**
#### Fixes #728

- [x] Ran `flake8`

#### Changes proposed in this pull request:

- testsuites/listener/shared/client_sg tests now run with a backing server instead of walrus
- More validation for database deletion
- conftest cleanup
- Fix a few races in tests due change from in memory walrus db
- exposed client.delete_database(url, name)
- Readme cleanup
- Resurrected client 2 sg test

TODO
- [x] net windows listener job update (client_sg, add 2 vms)
- [x] net windows listener job update (client_sg_topology_specific, add 2 vms)
- [x] add topo specific job for net windows

### CODE REVIEW

File: keywords/MobileRestClient.py [hunk 1]
```diff
@@ -351,40 +352,73 @@ def update_user(self, url, db, name, password=None, channels=None, roles=None):
...
+
+            if bucket_name is None:
+                raise keywords.exceptions.RestError("Creating database error. You must provide a couchbase server bucket name")
+            data["bucket"] = bucket_name
+
+            if sync_gateway_mode is None:
+                raise keywords.exceptions.RestError("You must specify either 'cc' or 'di' for sync_gateway_mode")
+
+            if sync_gateway_mode == "di" and index_bucket_name is None:
+                raise keywords.exceptions.RestError("You must an index_bucket_name if you are running in distributed index mode")
```
Comment (raghusarangapani) [inline keywords/MobileRestClient.py]: Typo, "provide" missing

File: keywords/MobileRestClient.py [hunk 2]
```diff
@@ -351,40 +352,73 @@ def update_user(self, url, db, name, password=None, channels=None, roles=None):
...
+
+            if sync_gateway_mode == "di" and is_index_writer is None:
+                raise keywords.exceptions.RestError("Make sure you provide whether you are targeting an index writer")
```
Comment (raghusarangapani) [inline keywords/MobileRestClient.py]: This log can be rephrased

File: testsuites/listener/shared/client_sg/test_autoprune.py [hunk 3]
```diff
@@ -172,6 +167,9 @@ def test_auto_prune_listener_keeps_conflicts_sanity(setup_client_syncgateway_tes
     sg_db_docs = client.add_docs(url=sg_url, db=sg_db, number=num_docs, id_prefix="doc", channels=sg_user_channels, auth=sg_session)
     assert len(sg_db_docs) == num_docs
 
+    # Wait for changes to be available on Sync Gateway
+    client.verify_docs_in_changes(url=sg_url, db=sg_db, expected_docs=sg_db_docs, auth=sg_session, polling_interval=1)
```
Comment (raghusarangapani) [inline testsuites/listener/shared/client_sg/test_autoprune.py]: Isn't polling interval too short?

Comment (sethrosetter) [inline testsuites/listener/shared/client_sg/test_autoprune.py]: This should be ok.

File: keywords/MobileRestClient.py (hunk 1 above)
Comment (sethrosetter) [inline keywords/MobileRestClient.py]: Done

File: keywords/MobileRestClient.py (hunk 2 above)
Comment (sethrosetter) [inline keywords/MobileRestClient.py]: Done

File: testsuites/listener/shared/client_sg/conftest.py [hunk 4]
```diff
@@ -23,7 +18,10 @@ def pytest_addoption(parser):
     parser.addoption("--liteserv-host", action="store", help="liteserv-host: the host to start liteserv on")
     parser.addoption("--liteserv-port", action="store", help="liteserv-port: the port to assign to liteserv")
     parser.addoption("--liteserv-storage-engine", action="store", help="liteserv-storage-engine: the storage engine to use with liteserv")
+    parser.addoption("--skip-provisioning", action="store_true", help="Skip cluster provisioning at setup", default=False)
     parser.addoption("--sync-gateway-version", action="store", help="sync-gateway-version: the version of sync_gateway to run tests against")
+    parser.addoption("--sync-gateway-mode", action="store", help="sync-gateway-mode: the version of sync_gateway to run tests against, channel_cache ('cc') or distributed_index ('di')")
```
Comment (borrrden) [inline testsuites/listener/shared/client_sg/conftest.py]: The "version" is odd.  The "mode" of sync_gateway would be better.

**Pull request merged**